src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -1136,6 +1139,98 @@ export const GlobalProvider = memo(
 
         const [connectingFrom, setConnectingFrom] = useState<OnConnectStartParams | null>(null);
 
+        const getNodesBoundingBox = useCallback(() => {
+            const nodes = getNodes().filter((n) => !n.parentNode);
+
+            const minX = Math.min(...nodes.map((n) => n.position.x));
+            const minY = Math.min(...nodes.map((n) => n.position.y));
+            const maxX = Math.max(...nodes.map((n) => n.position.x + (n.width ?? 0)));
+            const maxY = Math.max(...nodes.map((n) => n.position.y + (n.height ?? 0)));
+
+            return {
+                x: minX,
+                y: minY,
+                width: maxX - minX,
+                height: maxY - minY,
+            };
+        }, [getNodes]);
+
+        const downloadImage = (dataUrl: string, fileName: string) => {
+            const a = document.createElement('a');
+            a.href = dataUrl;
+            a.download = fileName;
+            document.body.appendChild(a);
+            a.click();
+            document.body.removeChild(a);
+        };
+
+        const [viewportExportPadding] = useViewportExportPadding;
+        const exportViewportScreenshot = useCallback(() => {
+            if (!reactFlowWrapper.current) return;
+
+            const oldViewport = currentReactFlowInstance.getViewport();
+
+            const reactFlowViewport = reactFlowWrapper.current.getBoundingClientRect();
+            const { x, y, width, height } = getNodesBoundingBox();
+            const paddedBoundingBox = {
+                x: x - viewportExportPadding,
+                y: y - viewportExportPadding,
+                width: width + viewportExportPadding * 2,
+                height: height + viewportExportPadding * 2,
+            };
+
+            const exportZoom = Math.min(
+                reactFlowViewport.width / paddedBoundingBox.width,
+                reactFlowViewport.height / paddedBoundingBox.height
+            );
+
+            currentReactFlowInstance.setViewport({
+                x: paddedBoundingBox.x * -1 * exportZoom,
+                y: paddedBoundingBox.y * -1 * exportZoom,
+                zoom: exportZoom,
+            });
+
+            const currentFlowWrapper = reactFlowWrapper.current;
+            if (!(currentFlowWrapper instanceof HTMLElement)) return;
```

Comment:
Could you move this check all the way up to where `if (!reactFlowWrapper.current) return;` is? That would allow us to remove the `if (!reactFlowWrapper.current) return;`.